test/shared.js

```diff
@@ -111,6 +111,21 @@ const checkResult = async (input, to, from, ethers, provider, errMsg) => {
     }
 };
 
+const waitForNumberOfBlock = async (provider, numberOfBlock) => {
+    const currentBlock = await provider.getBlockNumber();
+    let temp = await provider.getBlockNumber();
+    while (temp < currentBlock + numberOfBlock) {
+        if (network.name === 'hardhat') {
+            // Mine 1 block
+            await provider.send('evm_mine');
+        } else {
+            // wait 14 seconds
+            await new Promise((resolve) => setTimeout(resolve, 10000));
```

Comment:
it is not guaranteed 1 block, for Rinkeby it is 15 sec. 
Besides, the comment mismatches the code: 14 sec vs 10 sec